src/components/App/index.tsx

```diff
@@ -191,232 +48,37 @@ export default class App extends Component<Props, State> {
-        ...this.state,
-        source: { data, file, vectorImage },
-        loading: false,
-      };
-
-      // Default resize values come from the image:
-      for (const i of [0, 1]) {
-        newState = cleanMerge(newState, `images.${i}.preprocessorState.resize`, {
-          width: data.width,
-          height: data.height,
-          method: vectorImage ? 'vector' : 'browser-high',
-        });
-      }
-
-      this.setState(newState);
-    } catch (err) {
-      console.error(err);
-      this.showError('Invalid image');
-      this.setState({ loading: false });
-    }
-  }
-
-  async updateImage(index: number, options: UpdateImageOptions = {}): Promise<void> {
-    const { skipPreprocessing = false } = options;
-    const { source } = this.state;
-    if (!source) return;
-
-    // Each time we trigger an async encode, the counter changes.
-    const loadingCounter = this.state.images[index].loadingCounter + 1;
-
-    let images = cleanMerge(this.state.images, index, {
-      loadingCounter,
-      loading: true,
-    });
-
-    this.setState({ images });
-
-    const image = images[index];
-
-    let file: File | Fileish | undefined;
-    let preprocessed: ImageData | undefined;
-    let data: ImageData | undefined;
-    const cacheResult = this.encodeCache.match(source, image.preprocessorState, image.encoderState);
-
-    if (cacheResult) {
-      ({ file, preprocessed, data } = cacheResult);
-    } else {
-      try {
-        // Special case for identity
-        if (image.encoderState.type === identity.type) {
-          ({ file, data } = source);
-        } else {
-          preprocessed = (skipPreprocessing && image.preprocessed)
-            ? image.preprocessed
-            : await preprocessImage(source, image.preprocessorState);
-
-          file = await compressImage(preprocessed, image.encoderState, source.file.name);
-          data = await decodeImage(file);
-
-          this.encodeCache.add({
-            source,
-            data,
-            preprocessed,
-            file,
-            encoderState: image.encoderState,
-            preprocessorState: image.preprocessorState,
-          });
-        }
-      } catch (err) {
-        this.showError(`Processing error (type=${image.encoderState.type}): ${err}`);
-        throw err;
-      }
-    }
-
-    const latestImage = this.state.images[index];
-    // If a later encode has landed before this one, return.
-    if (loadingCounter < latestImage.loadedCounter) {
-      return;
-    }
-
-    images = cleanMerge(this.state.images, index, {
-      file,
-      data,
-      preprocessed,
-      downloadUrl: URL.createObjectURL(file),
-      loading: images[index].loadingCounter !== loadingCounter,
-      loadedCounter: loadingCounter,
-    });
-
-    this.setState({ images });
+  private async onIntroPickFile(file: File | Fileish) {
+    this.setState({ file });
   }
 
   @bind
-  showError (error: string) {
+  private showError(error: string) {
```

Comment:
Thought: could/should we lazyily load the snackbar too? it's not critical UI. Not sure if we're hoping this initial bundle payload is small enough to inline..

Reply:
Good idea. I'll see how much it adds to the bundle. Currently I think we should inline the CSS but not the JS (assuming we server render enough HTML for a meaningful render).

Reply:
CSS, JS
With this PR: 5.5k, 8.1k.
Without snackbar: 5.1k, 7.7k.

So that's a 0.8k saving, which we may lose by adding code to load the snackbar async (e.g. what if an error happens before the snackbar is ready).

I don't think it's worth it for the complexity jump. We'd save more by dropping the font (which I'm getting pretty tempted by).